partner_event/models/event_registration.py

```diff
@@ -0,0 +1,37 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, api
+
+
+class EventRegistration(models.Model):
+    _inherit = "event.registration"
+
+    def _prepare_partner(self, vals):
+        return {
+            'name': vals.get('name') or vals.get('email'),
+            'email': vals.get('email', False),
+            'phone': vals.get('phone', False),
+        }
+
+    @api.model
+    def create(self, vals):
+        if not vals.get('partner_id') and vals.get('email'):
+            partner_model = self.env['res.partner']
+            event_model = self.env['event.event']
+            partner_id = False
+            # Look for a partner with that email
+            partners = partner_model.search(
+                [('email', '=ilike', vals['email'])])
```

Comment:
If you use ilike and do not check the email, this happens:
![screenshot from 2015-11-10 10 16 40](https://cloud.githubusercontent.com/assets/973709/11058608/47695150-8794-11e5-8948-2495a1110823.png)
![screenshot from 2015-11-10 10 16 51](https://cloud.githubusercontent.com/assets/973709/11058609/492c1d06-8794-11e5-9ca6-d23578fd9696.png)